aiidalab_widgets_base/nodes.py

```diff
@@ -210,3 +264,88 @@ def find_node(self, pk):
             if getattr(node, "pk", None) == pk:
                 return node
         raise KeyError(pk)
+
+
+class _AppIcon:
+    def __init__(self, app, app_registry, path_to_root, node):
+
+        name = app["name"]
+        app_data = app_registry["apps"][name]
+        app_object = AiidaLabApp(name, app_data, AIIDALAB_APPS)
```

Comment:
Since you don't need a widget, it would probably be better to use the "internal" [`_AiidaLabApp` class](https://github.com/aiidalab/aiidalab/blob/45dfabb46f033c0d08db11a7a3ae435dcf7a5c01/aiidalab/app.py#L58). This one can also be much easier instantiated with [`.from_id(<app_name>)`](https://github.com/aiidalab/aiidalab/blob/45dfabb46f033c0d08db11a7a3ae435dcf7a5c01/aiidalab/app.py#L78). See the [`__main__` module](https://github.com/aiidalab/aiidalab/blob/45dfabb46f033c0d08db11a7a3ae435dcf7a5c01/aiidalab/__main__.py#L173) for examples.

Reply:
I would like to use the `is_installed` function that is only available in the `AiidaLabApp` class. In principle, if it is ok with you to move that function in the `_AiidaLabApp` class, then I can use it.